sql/delete.go

```diff
@@ -63,17 +63,36 @@ func (p *planner) Delete(n *parser.Delete) (planNode, error) {
 		if err := node.Err(); err != nil {
 			return nil, err
 		}
+		values := node.Values()
 
-		// TODO(tamird/pmattis): delete the secondary indexes too
-		primaryKey, err := encodeIndexKey(index, colMap, node.Values(), indexKey)
+		primaryIndexKeySuffix, err := encodeIndexKey(primaryIndex, colMap, values, nil)
 		if err != nil {
 			return nil, err
 		}
-		rowStartKey := proto.Key(primaryKey)
+		primaryIndexKey := append(primaryIndexKeyPrefix, primaryIndexKeySuffix...)
+
+		// Delete the secondary indexes.
+		for _, secondaryIndex := range tableDesc.Indexes[1:] {
+			secondaryIndexKeyPrefix := encodeIndexKeyPrefix(tableDesc.ID, secondaryIndex.ID)
+			secondaryIndexKey, err := encodeIndexKey(secondaryIndex, colMap, values, secondaryIndexKeyPrefix)
+			if err != nil {
+				return nil, err
+			}
+			rowStartKey := proto.Key(append(secondaryIndexKey, primaryIndexKeySuffix...))
+			b.DelRange(rowStartKey, rowStartKey.PrefixEnd())
```

Comment:
For secondary indexes, each index will correspond to a single key. It would be more efficient to just use `Del(rowStartKey)` here.